aws/s3.go

```diff
@@ -2,22 +2,25 @@ package aws
 
 import (
 	"fmt"
+	"math"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	awsgo "github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/s3"
 
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/gruntwork-io/cloud-nuke/logging"
 	"github.com/gruntwork-io/gruntwork-cli/errors"
 )
 
 // getS3BucketRegion returns S3 Bucket region.
-func getS3BucketRegion(svc *s3.S3, bucketName *string) (string, error) {
+func getS3BucketRegion(svc *s3.S3, bucketName string) (string, error) {
 	input := &s3.GetBucketLocationInput{
-		Bucket: bucketName,
+		Bucket: &bucketName,
```

Comment:
NIT: The idiom is to use `aws.String()` for converting args for the AWS SDK into pointers. Applies here and below.

Reply:
Done.